src/modules/identity/RadixRemoteIdentity.spec.ts

```diff
@@ -0,0 +1,71 @@
+import sinon from 'sinon'
+
+import { expect } from 'chai'
+import { describe, beforeEach, before } from 'mocha'
+
+import RadixServer from '../../../test/server/RadixServer'
+
+import {
+  radixUniverse,
+  RadixUniverse,
+  RadixAddress,
+  RadixRemoteIdentity,
+} from '../../index'
+
+let server
+let identity
+
+before(async () => {
+  // Bootstrap the universe
+  radixUniverse.bootstrap(RadixUniverse.LOCAL)
+
+  server = new RadixServer()
+  server.start()
+
+  identity = await RadixRemoteIdentity.createNew('dapp', 'dapp description', 'localhost', '54346')
+})
+
+describe('RadixRemoteIdentity', () => {
+
+  it('should return false if the Desktop wallet is closed', function (done) {
+    this.timeout(4000)
```

Comment:
I'm most likely asking a very stupid question here (the bad kind of "stupid", not the good kind), but what is `this` targeting in the above statement? Is it referring to the function (the second argument to `it(...)`) or the `it` itself? Does it even matter?

Is it the same thing as below example? (Should it be?)
```javascript
it('should return false...', (done) => {
    // test something
}).timeout(4000);
```

Reply:
Not stupid at all, the `this` keyword is one of the more tricky aspects of javascript. In old-style functions `function() {...}` `this` refers to the object from which the function is invoked. So here you don't even know what that is, because the `it` function just registers the test function, which is invoked later in a scope where this.timeout() is defined.

ES6 introduced arrow functions `() => {...}` where `this` instead gets bound to the scope of the function definition.

Your example is something else still, in that case you're trying to call `.timeout(4000)` on whatever the `it()` function returns (no idea what that is, likely `undefined`)

Here's some more info https://medium.com/quick-code/understanding-the-this-keyword-in-javascript-cb76d4c7c5e8

Reply:
Hey @laszlourszuly it's not a stupid question at all. In mocha arrow functions are discouraged, the lexical binding of the `this` value makes them unable to access the mocha context, and therefore will not work inside an arrow function.

But in this case you're right, the example you provided it's equivalent to what I did, so I'll probably update it since is neater. Thanks for that!

To do it globally it's possible to use something like this:

```
import mocha from 'mocha'

mocha.describe('foo', (suite) => {
  suite.timeout(100)

  suite.it('bar', () => ... ); 
})
```